javalib/src/main/scala/java/util/regex/Matcher.scala

```diff
@@ -0,0 +1,162 @@
+package java.util.regex
+
+import scalanative.native._, stdlib._, stdio._
+import scalanative.runtime.struct
+
+object RegMatch {
+  @struct
+  class regmatch_t(val rm_so: Int, val rm_eo: Int)
+}
+
+class FullRegMatch(val rm_so: Int,
+                   val rm_eo: Int,
+                   input: String,
+                   val retval: Int) {
+  def length: Int               = rm_eo - rm_so
+  def get: String               = input.substring(rm_so, rm_eo)
+  override def toString: String = s"$rm_so $rm_eo $input $retval"
```

Comment:
This is a really confusing toString. `s"FullRegMatch($rm_so, $rm_eo, $input, $retval)` would be a bit better.